homeassistant/components/media_player/kodi.py

```diff
@@ -75,12 +86,17 @@
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
     vol.Optional(CONF_TCP_PORT, default=DEFAULT_TCP_PORT): cv.port,
     vol.Optional(CONF_PROXY_SSL, default=DEFAULT_PROXY_SSL): cv.boolean,
+    vol.Optional(CONF_TURN_ON_ACTION, default=None): vol.In(TURN_ON_ACTION),
```

Comment:
We should not try to be smart. We should just support the Script syntax. They can do WoL if they want, or call anything else. Use `cv.SCRIPT_SCHEMA` for validation.

Once validates as SCRIPT_SCHEMA, just make it a script like this: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/alexa.py#L125-L126 and call that during turn_on